packages/wallet/ui/src/components/ChainConnector.jsx

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable react/display-name */
+import React, { useEffect, useState } from 'react';
+import clsx from 'clsx';
+import { makeStyles } from '@mui/styles';
+import LoadingButton from '@mui/lab/LoadingButton';
+import List from '@mui/material/List';
+import ListItem from '@mui/material/ListItem';
+import ListItemText from '@mui/material/ListItemText';
+import DialogTitle from '@mui/material/DialogTitle';
+import Dialog from '@mui/material/Dialog';
+import { NETWORK_CONFIGS, suggestChain } from '../util/SuggestChain';
+
+const useStyles = makeStyles(_ => ({
+  hidden: {
+    display: 'none',
+  },
+  connector: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    height: '100%',
+  },
+  centeredText: {
+    textAlign: 'center',
+  },
+  dialog: {
+    minWidth: 240,
+  },
+}));
+
+const ChainConnector = () => {
+  const classes = useStyles();
+  const [dialogOpened, setDialogOpened] = useState(false);
+  const [networkConfig, setNetworkConfig] = useState(null);
+  const [connectionInProgress, setConnectionInProgress] = useState(false);
+
+  const handleClose = () => {
+    setDialogOpened(false);
+  };
+
+  const selectNetworkConfig = nc => {
+    setNetworkConfig(nc);
+    setDialogOpened(false);
+  };
+
+  useEffect(() => {
+    if (!networkConfig) {
+      return () => {};
+    }
+
+    let networkChanged = false;
+    setConnectionInProgress(true);
+
+    const connect = async () => {
+      if (!window.getOfflineSigner || !window.keplr) {
+        setNetworkConfig(null);
+        alert('Please install the Keplr extension');
+      } else if (window.keplr.experimentalSuggestChain) {
+        try {
+          const cosmJS = await suggestChain(networkConfig[0]);
+          if (!networkChanged) {
+            setConnectionInProgress(false);
+            window.cosmJS = cosmJS;
```

Comment:
Setting this on the window so it can be tested through the console, but eventually the cosmJS ref should be stored in the application context.